internal/agent/agent.go

```diff
@@ -74,8 +74,24 @@ func GetAgentID(fileSystem afero.Fs) (string, error) {
 
 // Start the Agent. This will start the discovery ticker and the heartbeat ticker
 func (a *Agent) Start(ctx context.Context) error {
+	gathererRegistry := gatherers.NewRegistry(gatherers.StandardGatherers())
```

Comment:
Maybe we could move the `g, groupCtx` initialization after the  error checking, before ling `85`. As it won't be used if it fails before